rfd/xxxx-bot-for-cert-renewals.md

```diff
@@ -0,0 +1,298 @@
+omitted here for brevity):
+
+```
+$ tbot start --name=foo --auth-server=proxy.example.com:3080 --token=TOKEN ...
+```
+
+Note that both direct dialing or connecting through a Teleport proxy are supported.
+
+The bot joins the cluster and can be seen with the `tctl` command:
+
+```
+$ tctl bots ls
+ID              NAME
+0123-4567       foo
+```
+
+TODO: what else might we want to see? public addr? number (and kind) of certs?
+
+In the event of an incident, the bot can be locked with the new `tctl lock`
+functionality. A locked bot will be unable to renew certificates and will have
+its existing certificates revoked. (TODO: this is how it works, right?)
+
+Note that a bot can manage multiple sets of certificates, and locking a
+particular bot will apply to *all* of the certificates it manages.
+
+```
+$ tctl lock --bot=0123-4567
+```
+
+The confirmation message when running a lock command can include a list of the
+certificates the bot is managing.
+
+```
+This bot is managing the following certificates:
+
+- x509 server certificate at /etc/ssl (expires in 7 days)
+- SSH client certificate at /etc/ssh (expires in 8 hours)
+
+Locking the bot will prevent the renewal of these certificates.
+Are you sure you want to continue? (y/n)
+```
+
+#### Renewals
+
+There are two scenarios under which the bot will initiate a renewal:
+
+1. When a certificate is nearing expiry. A certificate is considered near
+   expiration if 75% of its TTL has elapsed, or when there are 4 hours or less
+   until expiration (whichever is sooner).
+2. When a user and/or host CA rotation is taking place.
+
+#### Certificate Specification
+
+
+The bot will be configured with one or more certificate specs, which can be
+supplied on the command line in the format of:
+
+```
+--cert=MODE,DESTINATION,RELOAD
+```
+
+*TODO*: how can a user specify DNS names to include in the certs?
+
+##### Mode
+
+The mode tells the bot which type of certificate to generate. 
+
+There are a number of ways the bot can be used to acquire and renew certificates.
+
+| Mode          | Certificate Type   | Signed By  |  Include User CA | Include Host CA |
+|---------------|--------------------|------------|------------------|-----------------|
+| `x509:client` | x509               | User CA    | no               | yes             |
+| `x509:server` | x509               | Host CA    | yes              | no              |
+| `ssh:client`  | SSH                | User CA    | no               | yes             |
+| `ssh:server`  | SSH                | Host CA    | yes              | no              |
+
+For example, for mode `x509:client`, the bot will issue x509 client certificates
+signed by Teleport's user CA. It also writes Teleport's host CA to the
+destination so that the client can be configured to trust the server.
+
+The mode also controls what the output of `tbot config` looks like.
+
+TODO:
+
+- do we need a mode equivalant to `tctl auth sign`'s `--compat` flag for openssh
```

Comment:
Not really, that should eventually go away, because it was targeting an ancient OpenSSH version that treated optional SSH cert extensions as critical by mistake.